Wheel_Speed_Algos.cpp

```diff
@@ -0,0 +1,89 @@
+#include <Arduino.h>
+
+#include "Wheel_Speed_Algos.h"
+
+const int STARBOARD_PIN = 2;
+const int PORT_PIN = 3;
+
+volatile unsigned int sboard_curr = 0;
+volatile unsigned int port_curr = 0;
+volatile unsigned int sboard_old = 0;
+volatile unsigned int port_old = 0;
+
+void sboard_click()
+{
+	sboard_old = sboard_curr;
+	sboard_curr = TCNT1;
+}
+
+void port_click()
+{
+	port_old = port_old;
+	port_curr= TCNT1;
+}
+
+//interrupt routine for timer2 overflow.
+//timer 2 is an 8bit timer with a 1024 scaling
+//Will run at 8MHz/256/1024 so ~30Hz
+//also am only pretty sure of 8MHz, if it's in fact 16 just some constant changing.
+
+//timer1 overflows with a freq. of 8MHz/65536/1024 so ~ 1.9 seconds
+//timer1 is at 125,000 Hz so one unit is 8*10^-6
+//one minute is 7,500,000 timer units
+#define MAGIC 7500000
+ISR(TIMER2_OVF_vect)
+{
+	//allow this interrupt to be interrupted
+	sei();
+	TCNT1 = 0x0000;
+	int sboard_delta = sboard_curr - sboard_old;
+	int port_delta = port_curr - port_old;
+	//make sure that it's the absolute difference regardless of overflows
+	//assuming 16bit integers. I'm pretty sure that's what this is.
+	sboard_delta = sboard_delta < 0 ? sboard_delta : (sboard_curr - (sboard_old - 0xFFFF));
+	port_delta = port_delta < 0 ? port_delta : (port_curr - (port_old - 0xFFFF));
+	//converts clicks into RPM
+	//proper math is that one click per tick times MAGIC (7,500,000) / teeth (22) will net RPM.
+	int sboard_rpm = MAGIC / sboard_delta / 22;
```

Comment:
Also be careful here: Arduino aliases `int` to int16_t rather than int32_t, so you'll overflow here. https://www.arduino.cc/en/Reference/Int

Use explicit int32_t tying instead (also good practice in general to avoid these kind of bugs).